introspection-engine/introspection-engine-tests/tests/rpc_calls/get_metadata_command_tests.rs

```diff
@@ -15,6 +15,18 @@ async fn metadata_for_mysql_should_work(api: &TestApi) -> TestResult {
     Ok(())
 }
 
+#[test_connector(tags(TiDB))]
+async fn metadata_for_tidb_should_work(api: &TestApi) -> TestResult {
+    setup(&api.barrel(), api.schema_name()).await?;
+
+    let result = api.get_metadata().await?;
+
+    assert_eq!(result.table_count, 3);
+    assert_eq!(result.size_in_bytes, 0);
```

Comment:
This value is not available in TiDB?

Reply:
I think this is related to the implementation of different storage engines.

The Prisma query engines get the value of `size_in_bytes` with the following SQL statement:

```sql
SELECT SUM(data_length + index_length) as size FROM information_schema.TABLES WHERE table_schema = ?
```

If we exceute the following SQL for MySQL and TiDB to check the metadata of tables created in the testcase:

```sql
SELECT TABLE_NAME, TABLE_ROWS, DATA_LENGTH, INDEX_LENGTH  FROM information_schema.TABLES WHERE table_schema = 'metadata_for_mysql_should_work';
```

We will get the result:

**For MySQL (InnoDB)**:

```
+------------+------------+-------------+--------------+
| TABLE_NAME | TABLE_ROWS | DATA_LENGTH | INDEX_LENGTH |
+------------+------------+-------------+--------------+
| Blog       | 0          | 16384       | 0            |
| Blog2      | 0          | 16384       | 0            |
| Blog3      | 0          | 16384       | 0            |
+------------+------------+-------------+--------------+
```

**For TiDB**:

```
+------------+------------+-------------+--------------+
| TABLE_NAME | TABLE_ROWS | DATA_LENGTH | INDEX_LENGTH |
+------------+------------+-------------+--------------+
| Blog       | 0          | 0           | 0            |
| Blog2      | 0          | 0           | 0            |
| Blog3      | 0          | 0           | 0            |
+------------+------------+-------------+--------------+
```

**For MySQL**

According to the official MySQL document:

> For InnoDB, DATA_LENGTH is the approximate amount of space allocated for the clustered index, in bytes. Specifically, it is the clustered index size, in pages, multiplied by the InnoDB page size.

$$
DATA\\_LENGTH = clustered\\_index\\_size \times innodb\\_page\\_size
$$

> InnoDB indexes are [B-tree](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_b_tree) data structures.

> The default size of an index page is 16KB. 

We can suppose that InnoDB's underlying storage will have a BTree root node even if there is no data, which will occupy 16Kb storage size, so we see `DATA_LENGTH = 16384` in the metadata.

**For TiDB**

The underlying layer of the TiDB data store uses the LSM Tree data structure, so `DATA_LENGTH` is calculated differently:

According to TIiDB official documents

> DATA_LENGTH: Data length. DATA_LENGTH = TABLE_ROWS * the sum of storage lengths of the columns in the tuple. The replicas of TiKV are not taken into account.

Because there are no data rows in the three tablesso we see `DATA_LENGTH = 0` in the metadata.

**Reference:** 

- https://dev.mysql.com/doc/refman/5.7/en/information-schema-tables-table.html
- https://dev.mysql.com/doc/refman/8.0/en/innodb-physical-structure.html
- https://docs.pingcap.com/tidb/stable/information-schema-tables#tables
- https://docs.pingcap.com/tidb/stable/tidb-storage
- https://docs.pingcap.com/tidb/dev/rocksdb-overview#rocksdb-overview

Reply:
Add comments for it: https://github.com/prisma/prisma-engines/pull/3308/commits/179b7f5c0a813dfe39e034e1d48d37877cc3ac95